src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -827,20 +826,16 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
                 contentHeaders = request.Content.Headers;
             }
 
-            string[] allHeaders = HeaderUtilities.DumpHeaders(request.Headers, contentHeaders)
-                .Split(s_headerDelimiters, StringSplitOptions.RemoveEmptyEntries);
+            // Prepare request headers
+            List<string> allHeaders = PrepareRequestHeaders(request.Headers, contentHeaders);
```

Comment:
Presumably setting up these headers is very common, and this has some allocations associated with it it'd be very nice to avoid: the params array being allocated to pass in the arguments, the ```List<string>``` allocated to store the results, any arrays allocated inside of the ```List<string>```, etc (the current code that iterates over the header values is also potentially allocating lots of strings inefficiently).  I think it'd be better if it were instead something like this:
```C#
if (request.Headers != null)
    AddRequestHeaders(request.Headers, retVal, rawHandle);
if (contentHeaders != null)
    AddRequestHeaders(contentHeaders, retVal, rawHandle);
}
...
private static void AddRequestHeaders(HttpHeaders headers, SafeCurlSlistHandle handle, ref IntPtr rawHandle)
{
    foreach (KeyValuePair<string, IEnumerable<string>> header in headers)
    {
        string headerStr = header.Key + ": " + string.Join(" ", header.Values);
        rawHandle = Interop.libcurl.curl_slist_append(rawHandle, headerStr);
        handle.SetHandle(rawHandle);
    }
}
```